tcms/xmlrpc/api/telemetry.py

```diff
@@ -0,0 +1,64 @@
+from modernrpc.core import rpc_method
+
+from tcms.management.models import Product
+from tcms.testcases.models import TestCase
+from tcms.testplans.models import TestPlan
+
+
+@rpc_method(name='Telemetry.get_data')
+def get_data(request):
+
+    test_plan_id = request.get('test_plan_id')
+    if not test_plan_id:
+        raise Exception('Required test_plan_id')
+
+    test_plan = TestPlan.objects.get(plan_id=test_plan_id)
+
+    test_cases = TestCase.objects.filter(plan=test_plan)
+
+    count = {
+        'manual': 0,
+        'automated': 0
+    }
+    for test_case in test_cases:
+        if test_case.is_automated:
+            count['manual'] += 1
+        else:
+            count['automated'] += 1
+    count['all'] = count['automated'] + count['manual']
```

Comment:
Try using aggregation instead of counting by hand. The database can do this much faster:
https://docs.djangoproject.com/en/2.2/topics/db/aggregation/#generating-aggregates-for-each-item-in-a-queryset

Reply:
Also you have a bug here. For the sample data that I use for testing I have a total of 8 test cases for the same product, which are all automated. The chart shows me all of them are manual.

Reply:
Actually, I don't think that aggregations will work in our case. The example you are referring is counting many to many relationships for each object in the query set, while what we have is just a foreign key to another table.

For the count of automated/manual we can do it on DB level easily:
```python
test_cases = TestCase.objects.filter(**query)

count = {
    'manual': test_cases.filter(is_automated=False).count(),
    'automated': test_cases.filter(is_automated=True).count(),
    'all': test_cases.count()
}
```
for category and priority count we can do it either by hand, or with something like:
```python
test_cases = TestCase.objects.filter(**query)

priorities = {}
for priority in Priority.objects.filter():
    priorities[priority.value] = test_cases.filter(priority=priority).count()
```